clang/labs/lab1/mortina_nastya/PrintikClass.cpp

```diff
@@ -0,0 +1,45 @@
+#include "clang/AST/ASTConsumer.h"
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Frontend/FrontendPluginRegistry.h"
+#include "llvm/Support/raw_ostream.h"
+
+class PrintClassFieldsVisitor
+    : public clang::RecursiveASTVisitor<PrintClassFieldsVisitor> {
+public:
+  bool VisitCXXRecordDecl(clang::CXXRecordDecl *declaration) {
+    if (declaration->isClass() || declaration->isStruct()) {
+      llvm::outs() << "Class Name: " << declaration->getNameAsString() << "\n";
+      for (auto field : declaration->fields()) {
+        llvm::outs() << "|_" << field->getNameAsString() << "\n";
+      }
+      llvm::outs() << "\n";
+    }
+    return true;
+  }
+};
+
+class PrintClassFieldsConsumer : public clang::ASTConsumer {
+public:
+  PrintClassFieldsVisitor Visitor;
+
+  void HandleTranslationUnit(clang::ASTContext &Context) override {
+    Visitor.TraverseDecl(Context.getTranslationUnitDecl());
+  }
+};
+
+class PrintClassFieldsAction : public clang::PluginASTAction {
+public:
+  std::unique_ptr<clang::ASTConsumer>
+  CreateASTConsumer(clang::CompilerInstance &CI, llvm::StringRef) override {
+    return std::make_unique<PrintClassFieldsConsumer>();
+  }
+
+  bool ParseArgs(const clang::CompilerInstance &CI,
+                 const std::vector<std::string> &Args) override {
+    return true;
```

Comment:
Please add an option that turns off fields printing, leaves struct name printing only.